pkg/http/api.go

```diff
@@ -2031,9 +2059,17 @@ func (a *api) onDirectActorMessage(reqCtx *fasthttp.RequestCtx) {
 		log.Debug(msg)
 		return
 	}
+	defer resp.Close()
+
 	invokev1.InternalMetadataToHTTPHeader(reqCtx, resp.Headers(), reqCtx.Response.Header.Set)
-	contentType, body := resp.RawData()
-	reqCtx.Response.Header.SetContentType(contentType)
+	body, err := resp.RawDataFull()
+	if err != nil {
+		msg := NewErrorResponse("ERR_ACTOR_INVOKE_METHOD", fmt.Sprintf(messages.ErrActorInvoke, err))
+		respond(reqCtx, withError(fasthttp.StatusInternalServerError, msg))
+		log.Debug(msg)
+		return
```

Comment:
please add UT for this

Reply:
I am afraid there's no way to write a UT for this, as an error here would never happen in Dapr today.

Because we use fasthttp which doesn't support reading data as a stream, we are populating the request with a byte slice:

https://github.com/ItalyPaleAle/dapr/blob/aa229cc673e74f4d5bb03846ef0416c4a8c566aa/pkg/http/api.go#L2027-L2028

Because of that, I cannot create a unit test that uses a stream that returns an error in the stream inside the request body (like I did for the gRPC channel).